includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -1909,10 +1927,10 @@ protected function guess_modal_role( DOMElement $modal ) {
 			return 'dialog';
 		}
 
-		$classes = $modal->getAttribute( 'class' );
+		$classes = explode( ' ', $modal->getAttribute( 'class' ) );
 
-		foreach ( [ 'navigation', 'menu', 'search', 'alert', 'figure', 'form', 'img', 'toolbar', 'tooltip' ] as $role ) {
-			if ( false !== strpos( $classes, $role ) ) {
+		foreach ( self::$modal_roles as $role ) {
+			if ( in_array( $role, $classes, true ) ) {
```

Comment:
Made this more robust, so a class such as `top-navigation` would not match when comparing against `navigation`.